client/src/types/interfaces.ts

```diff
@@ -1,3 +1,77 @@
+export interface ProductData {
+  id: number;
+  attributes: {
+    title: string;
+    disc: string;
+    price: number;
+    img: {
+      data: {
+        attributes: {
+          url: string;
+        };
+      };
+    };
+  };
+}
+
+export interface ProductsType {
+  numberOfProducts: number;
+}
+
+export interface CategoryType {
+  numberOfCategories: number;
+  filter: string;
+}
+
+export interface CategoryData {
+  id: number;
+  attributes: {
+    title: string;
+    img: {
+      data: {
+        attributes: {
+          url: string;
+        };
+      };
+    };
+  };
+}
+
+export interface SaleCardProps {
+  backgroundImage: string;
+  title: string;
+  subtitle: string;
+  discount: string;
+}
+
+export interface IconButtonProps {
+  icon: string;
+}
+
+export interface IconButtonWithTextProps {
+  icon: string;
+  text: string;
+  buttonColor: 'purple' | 'white';
+}
+
+export interface SectionTitleProps {
+  text: string;
+}
+
+export interface TextButtonProps {
+  text: string;
+  buttonColor: 'purple' | 'white';
+}
+
+export interface Product {
+  id: string;
+  price: number;
+  quantity: number;
+}
+
+export interface CartProps {
```

Comment:
see the comment for Component interfaces
(interfaces for components should stay in that folder 
./interfaces.ts
or just placeed above function
)